In [1]:
# importing the necessary libraries.
import pandas as pd
import numpy as np
import seaborn as sb
import matplotlib.pyplot as plt
%matplotlib inline


In [2]:
# loading the dataset
data=pd.read_csv(r"C:\Users\HP\Downloads\Bengaluru_House_Data.csv")

In [3]:
data.head(5)

area_type   availability                  location       size  \
0  Super built-up  Area         19-Dec  Electronic City Phase II      2 BHK   
1            Plot  Area  Ready To Move          Chikka Tirupathi  4 Bedroom   
2        Built-up  Area  Ready To Move               Uttarahalli      3 BHK   
3  Super built-up  Area  Ready To Move        Lingadheeranahalli      3 BHK   
4  Super built-up  Area  Ready To Move                  Kothanur      2 BHK   

   society total_sqft  bath  balcony   price  
0  Coomee        1056   2.0      1.0   39.07  
1  Theanmp       2600   5.0      3.0  120.00  
2      NaN       1440   2.0      3.0   62.00  
3  Soiewre       1521   3.0      1.0   95.00  
4      NaN       1200   2.0      1.0   51.00

In [4]:
# making copy of data .
data_copy= data.copy()

In [5]:
data_copy.sample(10)

area_type   availability                  location  \
4279             Plot  Area  Ready To Move               Hegde Nagar   
12670        Built-up  Area  Ready To Move               Hennur Road   
8627             Plot  Area  Ready To Move             Kenchenahalli   
7066   Super built-up  Area  Ready To Move               Uttarahalli   
3510   Super built-up  Area         18-Jul  Electronic City Phase II   
6203   Super built-up  Area  Ready To Move        7th Phase JP Nagar   
8339   Super built-up  Area  Ready To Move               Rajiv Nagar   
8458   Super built-up  Area  Ready To Move             Iblur Village   
12607  Super built-up  Area  Ready To Move          Ramagondanahalli   
5183         Built-up  Area  Ready To Move                   Kengeri   

            size  society total_sqft  bath  balcony   price  
4279   2 Bedroom      NaN       1200   2.0      0.0   75.00  
12670      2 BHK  Eltert        1450   2.0      2.0   80.00  
8627   3 Bedroom      NaN        480   3.0      3.0   65.00  
7066       3 BHK  SLnssPr       1330   2.0      2.0   39.90  
3510       3 BHK  Sryalan       1400   2.0      2.0   41.58  
6203       3 BHK  Eladero       1675   2.0      2.0  135.00  
8339       4 BHK  PuoodSk       2340   4.0      1.0  145.00  
8458       3 BHK  Emine P       2666   4.0      1.0  200.00  
12607      3 BHK  Srdor S       1680   2.0      NaN   68.00  
5183       1 BHK      NaN        502   1.0      1.0   25.00

In [6]:
# Checking the shape of data
data_copy.shape

(13320, 9)

* We have 13320 rows and 9 columns.

In [7]:
# checking the info
data_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13320 entries, 0 to 13319
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   area_type     13320 non-null  object 
 1   availability  13320 non-null  object 
 2   location      13319 non-null  object 
 3   size          13304 non-null  object 
 4   society       7818 non-null   object 
 5   total_sqft    13320 non-null  object 
 6   bath          13247 non-null  float64
 7   balcony       12711 non-null  float64
 8   price         13320 non-null  float64
dtypes: float64(3), object(6)
memory usage: 936.7+ KB


* There are 3 columns based on float datatype and 6 columns based on object datatype.

In [8]:
# checking for null values 
data_copy.isnull().sum()

area_type          0
availability       0
location           1
size              16
society         5502
total_sqft         0
bath              73
balcony          609
price              0
dtype: int64

* We have null values in our dataset.Which have to removed or replaced by its mean or mod values.

In [9]:
data_copy.columns

Index(['area_type', 'availability', 'location', 'size', 'society',
       'total_sqft', 'bath', 'balcony', 'price'],
      dtype='object')

* List of a columns

In [10]:
data_copy.describe().T

count        mean         std  min   25%   50%    75%     max
bath     13247.0    2.692610    1.341458  1.0   2.0   2.0    3.0    40.0
balcony  12711.0    1.584376    0.817263  0.0   1.0   2.0    2.0     3.0
price    13320.0  112.565627  148.971674  8.0  50.0  72.0  120.0  3600.0

# cleaning the columns one by one 

### (1) for columns area_type

In [11]:
data_copy["area_type"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 13320 entries, 0 to 13319
Series name: area_type
Non-Null Count  Dtype 
--------------  ----- 
13320 non-null  object
dtypes: object(1)
memory usage: 104.2+ KB


* The data stored in a area_type column is of object datatype.

In [12]:
# checking for unique values.
data_copy["area_type"].unique()

array(['Super built-up  Area', 'Plot  Area', 'Built-up  Area',
       'Carpet  Area'], dtype=object)

In [13]:
data_copy["area_type"].nunique()

4

* There are total 4 unique values present in area_type column.

In [14]:
# counting the values of area_type column
data_copy["area_type"].value_counts()

area_type
Super built-up  Area    8790
Built-up  Area          2418
Plot  Area              2025
Carpet  Area              87
Name: count, dtype: int64

* There are 8790 counts for Super built-up Area.
* There are 2418 counts for Built-up Area.
* There are 2025 counts for Plot Area.
* There are 87 counts for Carpet Area.

### (2).availability column


In [15]:
# checking the info
data_copy["availability"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 13320 entries, 0 to 13319
Series name: availability
Non-Null Count  Dtype 
--------------  ----- 
13320 non-null  object
dtypes: object(1)
memory usage: 104.2+ KB


* The data in availability column is of object datatype

In [16]:
# checking the unique values 
data_copy["availability"].unique()

array(['19-Dec', 'Ready To Move', '18-May', '18-Feb', '18-Nov', '20-Dec',
       '17-Oct', '21-Dec', '19-Sep', '20-Sep', '18-Mar', '20-Feb',
       '18-Apr', '20-Aug', '18-Oct', '19-Mar', '17-Sep', '18-Dec',
       '17-Aug', '19-Apr', '18-Jun', '22-Dec', '22-Jan', '18-Aug',
       '19-Jan', '17-Jul', '18-Jul', '21-Jun', '20-May', '19-Aug',
       '18-Sep', '17-May', '17-Jun', '21-May', '18-Jan', '20-Mar',
       '17-Dec', '16-Mar', '19-Jun', '22-Jun', '19-Jul', '21-Feb',
       'Immediate Possession', '19-May', '17-Nov', '20-Oct', '20-Jun',
       '19-Feb', '21-Oct', '21-Jan', '17-Mar', '17-Apr', '22-May',
       '19-Oct', '21-Jul', '21-Nov', '21-Mar', '16-Dec', '22-Mar',
       '20-Jan', '21-Sep', '21-Aug', '14-Nov', '19-Nov', '15-Nov',
       '16-Jul', '15-Jun', '17-Feb', '20-Nov', '20-Jul', '16-Sep',
       '15-Oct', '15-Dec', '16-Oct', '22-Nov', '15-Aug', '17-Jan',
       '16-Nov', '20-Apr', '16-Jan', '14-Jul'], dtype=object)

In [17]:
data_copy["availability"].nunique()

81

* There are total 81 unique values in it.

In [18]:
# checking for data which are ready to move
data_copy[data_copy["availability"]=="Ready To Move"]

area_type   availability               location       size  \
1                Plot  Area  Ready To Move       Chikka Tirupathi  4 Bedroom   
2            Built-up  Area  Ready To Move            Uttarahalli      3 BHK   
3      Super built-up  Area  Ready To Move     Lingadheeranahalli      3 BHK   
4      Super built-up  Area  Ready To Move               Kothanur      2 BHK   
5      Super built-up  Area  Ready To Move             Whitefield      2 BHK   
...                     ...            ...                    ...        ...   
13314  Super built-up  Area  Ready To Move      Green Glen Layout      3 BHK   
13315        Built-up  Area  Ready To Move             Whitefield  5 Bedroom   
13316  Super built-up  Area  Ready To Move          Richards Town      4 BHK   
13317        Built-up  Area  Ready To Move  Raja Rajeshwari Nagar      2 BHK   
13319  Super built-up  Area  Ready To Move           Doddathoguru      1 BHK   

       society total_sqft  bath  balcony  price  
1      Theanmp       2600   5.0      3.0  120.0  
2          NaN       1440   2.0      3.0   62.0  
3      Soiewre       1521   3.0      1.0   95.0  
4          NaN       1200   2.0      1.0   51.0  
5      DuenaTa       1170   2.0      1.0   38.0  
...        ...        ...   ...      ...    ...  
13314  SoosePr       1715   3.0      3.0  112.0  
13315  ArsiaEx       3453   4.0      0.0  231.0  
13316      NaN       3600   5.0      NaN  400.0  
13317  Mahla T       1141   2.0      1.0   60.0  
13319      NaN        550   1.0      1.0   17.0  

[10581 rows x 9 columns]

* There are total 10581 houses in bangaluru which are ready to move.


In [19]:
# checking  for Immediate Possession 
data_copy[data_copy["availability"]=='Immediate Possession']

area_type          availability        location size  society  \
579    Plot  Area  Immediate Possession  Sarjapur  Road  NaN  Asiss B   
1775   Plot  Area  Immediate Possession        IVC Road  NaN  Orana N   
2264   Plot  Area  Immediate Possession    Banashankari  NaN      NaN   
2809   Plot  Area  Immediate Possession  Sarjapur  Road  NaN  AsdiaAr   
2862   Plot  Area  Immediate Possession     Devanahalli  NaN  Ajleyor   
5333   Plot  Area  Immediate Possession     Devanahalli  NaN  Emngs S   
6423   Plot  Area  Immediate Possession      Whitefield  NaN  SRniaGa   
6636   Plot  Area  Immediate Possession          Jigani  NaN  S2enste   
6719   Plot  Area  Immediate Possession         Hoskote  NaN  SJowsn    
7680   Plot  Area  Immediate Possession    Kasavanhalli  NaN      NaN   
8306   Plot  Area  Immediate Possession          Anekal  NaN  Vaensnd   
8565   Plot  Area  Immediate Possession      Carmelaram  NaN  Ocezes    
8703   Plot  Area  Immediate Possession     Mysore Road  NaN  Naoden    
10634  Plot  Area  Immediate Possession      Rajankunte  NaN      NaN   
11019  Plot  Area  Immediate Possession       Yelahanka  NaN  Saaveha   
11569  Plot  Area  Immediate Possession      Hosur Road  NaN  AVeldun   

        total_sqft  bath  balcony    price  
579    1200 - 2400   NaN      NaN   34.185  
1775   2000 - 5634   NaN      NaN  124.000  
2264          2400   NaN      NaN  460.000  
2809   1200 - 2400   NaN      NaN   28.785  
2862   1500 - 2400   NaN      NaN   46.800  
5333   2100 - 5405   NaN      NaN  177.115  
6423          2324   NaN      NaN   26.730  
6636          1500   NaN      NaN   25.490  
6719    800 - 2660   NaN      NaN   28.545  
7680          5000   NaN      NaN  400.000  
8306          1453   NaN      NaN   16.500  
8565          2000   NaN      NaN  120.000  
8703   1200 - 2400   NaN      NaN   42.300  
10634         1575   NaN      NaN   31.110  
11019  1200 - 1800   NaN      NaN   12.750  
11569         1350   NaN      NaN    8.440

* There are 16 houses which are under immediate possession.

In [20]:
# checking for null value
data_copy["availability"].isnull().sum()

0

*There is no null values present in it.

### (3) For location column

In [21]:
# checking it info
data_copy["location"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 13320 entries, 0 to 13319
Series name: location
Non-Null Count  Dtype 
--------------  ----- 
13319 non-null  object
dtypes: object(1)
memory usage: 104.2+ KB


* The data present in this columnnis of object datatype.

In [22]:
# checking fo unique values
data_copy["location"].unique()

array(['Electronic City Phase II', 'Chikka Tirupathi', 'Uttarahalli', ...,
       '12th cross srinivas nagar banshankari 3rd stage',
       'Havanur extension', 'Abshot Layout'], dtype=object)

In [23]:
data_copy["location"].nunique()

1305

* There are 1305 unique values present in it.

In [24]:
# checking for the null values.
data_copy["location"].isnull().sum()

1

* There is 1 null value present it.

In [25]:
# As the data is of object datatype here so we are replacing the null value with its mode value .
mode=data_copy["location"].mode()[0]

In [26]:
mode

'Whitefield'

In [27]:
# filling the null value with its mode .
data_copy["location"].fillna('Whitefield',inplace=True)


* The null value is replace with its mode value.

In [28]:
# cross checking for the null value 
data_copy["location"].isnull().sum()

0

* successfully we have replaced the null value with mode value,i.e., 'Whitefield'.